File: src/main/java/entity/Booking.java
package entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Booking {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer bookingId;
    
    private LocalDateTime bookingDate = LocalDateTime.now();
    private Integer noOfPax;
    private Double tourAmount;
    private Double taxes;
    private Double totalAmount;
    private String paymentStatus;
    
    @ManyToOne
    @JoinColumn(name = "cust_id")
    private Customer customer;
    
    @ManyToOne
    @JoinColumn(name = "package_id")
    private Package tourPackage;  
    
    @ManyToOne
    @JoinColumn(name = "departure_id")
    private Departure departure;
    
    @OneToMany(mappedBy = "booking", cascade = CascadeType.ALL)
    private List<Passenger> passengers;
}

File: DotNet_Backend/SafarnamaApplication/SafarnamaApplication/Models/Booking.cs
using System;
using System.Collections.Generic;

namespace SafarnamaApplication.Models;

public partial class Booking
{
    public int BookingId { get; set; }

    public DateTime? BookingDate { get; set; }

    public int? CustId { get; set; }

    public int? PackageId { get; set; }

    public int? DepartureId { get; set; }

    public int? NoOfPax { get; set; }

    public double? TourAmount { get; set; }

    public double? Taxes { get; set; }

    public double? TotalAmount { get; set; }

    public string? PaymentStatus { get; set; }

    public virtual Customer? Cust { get; set; }

    public virtual Departure? Departure { get; set; }

    public virtual Package? Package { get; set; }

    public virtual ICollection<Passenger> Passengers { get; set; } = new List<Passenger>();
}
